### PR TITLE
Read git auth token from backend config

### LINKED ISSUE #2991: TechDocs Backend: Read GHE token from configuration instead of environment variables
**(I was not sure if this should be considered as a bug)**

For some reason, the access token that is used to while cloning the repo is read from the environment variables instead of from the configuration reader.
https://github.com/spotify/backstage/blob/b46863b88b933b79ec74af2550a0b4d6e473a9a2/plugins/techdocs-backend/src/helpers.ts#L130

## Possible Implementation
Pass config reader as an additional parameter and read the token from there.

## Context
I was trying to enable TechDocs with GHE setup.

### PR DESCRIPTION
**Pull request by steelrooter**
## Hey, I just made a Pull Request!

While testing TechDocs from an internal GitLab repo, we found that the auth mechanism is not working properly and we were getting `Error: remote authentication required but no callback set`. Turned out that the token provided under `integrations.gitlab` wasn't being used and instead the GitlabPreparer was dependent on `catalog.processors.gitlab.privateToken` or the env var `process.env.GITLAB_TOKEN`. This PR adds support to check for the token first in the `integrations` config and then falls back to old behaviour. We have tested this with our internal GitLab.

#### Points to note
1. I am not sure of how GitHub or Azure auth logic is yet. If someone can verify it, that would be great.
2. It's my first major contribution here and first time working with TypeScript (or JavaScript). So please forgive my mistakes. Give me feedback and I will incorporate it :)

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by hooloovooo** [comment]
Not sure how we should do with the legacy token config. Is it better just not supporting it, or do we want a transition period?

**Comment by roflc0pter** [comment]
This is great PR. I was struggling with the same issues while setting up GHE for TechDocs.

**Comment by OrkoHunter** [comment]
Let's keep the backwords compatibility for config now. cc @hooloovooo

**Review by hooloovooo — APPROVED** [review]

**Review by OrkoHunter — APPROVED** [review]
@steelrooter Grateful for your work on this PR. Let's ship it once merge conflicts are fixed. Welcome to the community. 🥳

**Comment by steelrooter** [comment]
Resolving them now. Curious to know if the community prefers a "squash and force push" or multiple commits approach in PRs.

**Comment by steelrooter** [comment]
I followed the "rebase, squash and force push" strategy, as I am used to it in my org. Let me know if the community prefers the multiple commits strategy.

**Comment by hooloovooo** [comment]
Nice work @steelrooter ! Thanks for the pull request and welcome! :)

**Pull request merged**
